### PR TITLE
DAML-LF forbid non-generic comparison in 1.dev

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**
In this PR we restrict the usage of non generic comparison builtins in LF >= 1.dev

This PR advances the states of #2256 

### Pull Request Checklist

- [X] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [X] Include appropriate tests
- [X] Set a descriptive title and thorough description
- [X] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Comment by remyhaemmerle-da** [comment]
blocked by #5021

File: daml-lf/archive/src/main/scala/com/digitalasset/daml/lf/archive/DecodeV1.scala [hunk 1]
```diff
@@ -1434,15 +1453,30 @@ private[lf] object DecodeV1 {
...
+        minVersion = partyOrdering,
+        maxVersion = Some(genComparison),
         implicitParameters = List(TParty),
-        minVersion = partyOrdering),
-      BuiltinFunctionInfo(LESS_INT64, BLess, implicitParameters = List(TInt64)),
+      ),
+      BuiltinFunctionInfo(
+        LESS_INT64,
+        BLess,
+//        maxVersion = Some(genComparison),
```
Comment (associahedron) [inline daml-lf/archive/src/main/scala/com/digitalasset/daml/lf/archive/DecodeV1.scala]: Is there a reason for this one to be commented out?

Comment (remyhaemmerle-da) [reply daml-lf/archive/src/main/scala/com/digitalasset/daml/lf/archive/DecodeV1.scala]: I was debugging the issue you fix in #5025.
I uncommented.

**Comment by associahedron** [comment]
All the blockers are merged

**Pull request merged**